### PR TITLE
Make cfdot available in a properly shared location.

### PR DESCRIPTION
**Pull request by andreas-kupries**
## Description

Patched cfdot job prestart to a shared location accessible in all the container of an instance group. The patches are all the same for the three affected instance groups. (As the job itself is the same).

Added comments to the diego-cell patch setup as a reminder to maintainers as to the intent of the complexity.

## Motivation and Context

See ticket #738 

## How Has This Been Tested?

Testing was done in a local minikube deployment. After the cluster became :heavy_check_mark: all affected instance groups were entered into via `kubectl exec`, the `cfdot.sh` was sourced from its new location and then a few `cfdot` commands invoked (`--help`, `locks`, `task`, `domains`, `cells`, `presences`.).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: deploy/helm/kubecf/assets/operations/instance_groups/diego-cell.yaml [hunk 1]
```diff
@@ -225,10 +225,12 @@
   value: true
 
 {{- if .Values.features.suse_buildpacks.enabled }}
+# For SUSE buildpacks take all patches
```
Comment (jandubois) [inline deploy/helm/kubecf/assets/operations/instance_groups/diego-cell.yaml:228]: The check really should be against `features.suse_default_stack.enabled`, but that is outside the scope of this PR, and besides there are many other locations with this issue.

File: bosh/releases/pre_render_scripts/diego-api/cfdot/jobs/patch_pre-start.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+##
+# ATTENTION: This is part of a set of six interconnected patches.
+#            Two files spread over three instance groups.
```
Comment (jandubois) [inline bosh/releases/pre_render_scripts/diego-api/cfdot/jobs/patch_pre-start.sh:4]: I wish these patches were somehow implemented with a template helper function, so we don't have almost identical code in 3 different places. But this is also true for other patches we have, e.g. the buildpacks, that move files to `/var/vcap/data/shared-packages`. So maybe we can consolidate all this later (hah) in a future cleanup PR.

File: deploy/helm/kubecf/assets/operations/instance_groups/diego-cell.yaml (hunk 1 above)
Comment (andreas-kupries) [reply deploy/helm/kubecf/assets/operations/instance_groups/diego-cell.yaml:228]: Leaving it to separate PR

File: bosh/releases/pre_render_scripts/diego-api/cfdot/jobs/patch_pre-start.sh (hunk 2 above)
Comment (andreas-kupries) [reply bosh/releases/pre_render_scripts/diego-api/cfdot/jobs/patch_pre-start.sh:4]: Leaving it to later PR.
